analyzer/queries/queries.go

```diff
@@ -1,22 +1,89 @@
 package queries
 
 const (
-	LatestBlock = `
+	LatestProcessedBlock = `
     SELECT height FROM chain.processed_blocks
-      WHERE analyzer = $1
+      WHERE analyzer = $1 AND processed_time IS NOT NULL
     ORDER BY height DESC
     LIMIT 1`
 
+	FirstUnprocessedBlock = `
+    SELECT height FROM chain.processed_blocks
+      WHERE analyzer = $1 AND processed_time IS NULL
+    ORDER BY height
+    LIMIT 1`
+
+	UnlockBlockForProcessing = `
+    UPDATE chain.processed_blocks
+    SET locked_time = NULL
+    WHERE analyzer = $1 AND height = $2`
+
+	PickBlocksForProcessing = `
+    -- Locks a range of unprocessed blocks for an analyzer, handling expired locks and updating the lock status.
+    -- Parameters:
+    -- $1 = analyzer name (text)
+    -- $2 = minimum block height (integer)
+    -- $3 = maximum block height (integer, optional)
+    -- $4 = lock timeout in minutes (integer)
+    -- $5 = number of blocks to lock (integer)
+    BEGIN; -- Start a transaction, so that 'FOR UPDATE' locking persists for the whole query.
```

Comment:
Yeah, after looking through the docs some more, I agree that it is not needed. Should be roughly as you described. 

- FOR UPDATE locks until the current transaction ends
- In case of no explicit transactions, each statement is implicitly executed in a transaction
- in this case, there is a single statement (INSERT), with [_attached sub-statements (CTEs)_](https://www.postgresql.org/docs/current/queries-with.html)